java-api-core/test/com/kinvey/java/BaseUserTest.java

```diff
@@ -314,5 +318,28 @@ public void testMICAPIVersionAppendsV() throws IOException{
         assertEquals(getClient().getMICApiVersion(), "v2");
     }
 
+    public void testClientAppVersionHeader() throws IOException, NoSuchFieldException, IllegalAccessException {
+        String clientAppVersion = "1.2.3";
+        getClient().setClientAppVersion(clientAppVersion);
+        UserStoreRequestManager user = new UserStoreRequestManager(getClient(), createBuilder(getClient()));
+        assertNotNull(user);
+        UserStoreRequestManager.LoginRequest loginRequest = user.createBlocking("test_name", "test_login").buildAuthRequest();
+        Field kinveyAuthRequestField = loginRequest.getClass().getDeclaredField("request"); //NoSuchFieldException
+        kinveyAuthRequestField.setAccessible(true);
+        KinveyAuthRequest request = (KinveyAuthRequest) kinveyAuthRequestField.get(loginRequest);
+        Field kinveyHeadersField = request.getClass().getDeclaredField("kinveyHeaders"); //NoSuchFieldException
+        kinveyHeadersField.setAccessible(true);
+        KinveyHeaders kinveyHeaders = (KinveyHeaders) kinveyHeadersField.get(request);
+        String clientAppVersionHeader = (String) kinveyHeaders.get("X-Kinvey-Client-App-Version");
+        assertEquals(clientAppVersion, clientAppVersionHeader);
+    }
+
```

Comment:
Please also add a test case to verify that if **clientAppVersion** is not set, that the **X-Kinvey-Client-App-Version** header is not added to the request.

Reply:
Added